mozilla/foundation.mozilla.org
# Updated required fields and verbose names

# Description



Related PRs/issues: #9464

Per request of the design team, I have implemented the following changes to the "Promote" panel:

Sitewide: 
- [x] Update name of `Search image:` to `Share image:` since this is a more accurate discretion 
- [x] Add hint text under `Share image:` that says "Image must be high quality, include our logo mark and have the dimensions 1200 x 628 px. For more design guidelines [see here.](https://foundation.mozilla.org/docs/brand/brand-identity/social-media/#og-images)"

Blog only: (not sitewide since some pages inherit from their parent page)
- [x] Make the `Share image:` a required field on all blog post pages
- [x] Make `Meta description:` a required field
- [x] Remove the `Show in menus:` since it  doesn't actually do anything for blog posts

# Notes

There was some discussion in the [original ticket](https://github.com/mozilla/foundation.mozilla.org/issues/7638), where Pomax mentions that the "Promote" tab is a core wagatail feature, and it might require a large lift to update, if it is even possible. ([Link to comment](https://github.com/mozilla/foundation.mozilla.org/issues/7638#issuecomment-965650660))

However, I was able to use [this article](https://learnwagtail.com/tutorials/customizing-default-property-values/) written by Kalob for reference, as he provides a way to override things such as help text and verbose names of these wagtail default fields.

**Edit:**
Unfortunately I found that updating the blog pages meta at the end of the file would not work, as  using the `PageName._meta.get_field()` method does not only update the desired pages fields, but the fields of every page in the DB!

In order to maintain the desired functionality of having the share_image and share_description fields set as required for Blog Pages, I instead made this addition to the BlogPage's "[BlogPageForm](https://github.com/mozilla/foundation.mozilla.org/blob/da8b6fb62a2bddab24913644d481b502f5d702e3/network-api/networkapi/wagtailpages/forms.py#L29)" which works nicely while not affecting any other pages.


# Steps to test

1. Check this branch out and run `inv new-db` if needed
2. Visit the CMS and visit the home page, or any other page that is **not** a blog page
3. Click on the promote panel
4. Note that the `search_image` field is now `share_image` with updated help text, and that neither share_image or search_description are required.

![Screenshot 2022-10-13 at 19-59-58 Wagtail - Editing Homepage Homepage](https://user-images.githubusercontent.com/18314510/195752551-7b2a4351-d6a1-478a-9437-8108eb3210b1.png)

5. Visit a blog page, and click on the promote tab

6. Note that the `search_image` field is now updated to `share_image` with updated help text, and now is required alongside the description field. Also note that the `show_in_menus` field is not being rendered.

![Screenshot 2022-10-13 at 20-02-23 Wagtail - Editing Blog page Initial test blog post with fixed title](https://user-images.githubusercontent.com/18314510/195752920-9e0bc5f6-7058-4f6b-93e6-50ae9c06c425.png)

7. If everything is looking/working as expected, testing is complete!
